msal/broker.py

```diff
@@ -0,0 +1,198 @@
+    if prompt:
+        if prompt == "select_account":
+            if login_hint:
+                # FWIW, AAD's browser interactive flow would honor select_account
+                # and ignore login_hint in such a case.
+                # But pymsalruntime 0.3.x would pop up a meaningless account picker
+                # and then force the account_hint user to re-input password. Not what we want.
+                # https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1744492
+                login_hint = None  # Mimicing the AAD behavior
+                logger.warning("Using both select_account and login_hint is ambiguous. Ignoring login_hint.")
+            params.set_select_account_option(
+                pymsalruntime.SelectAccountOption.SHOWLOCALACCOUNTSCONTROL)
+        else:
+            logger.warning("prompt=%s is not supported by this module", prompt)
+    for k, v in kwargs.items():  # This can be used to support domain_hint, max_age, etc.
+        if v is not None:
+            params.set_additional_parameter(k, str(v))  # TODO: End-to-end test
+    if claims:
+        params.set_decoded_claims(claims)
+    callback_data = _CallbackData()
+    pymsalruntime.signin_interactively(
+        window or pymsalruntime.get_console_window() or pymsalruntime.get_desktop_window(),  # Since pymsalruntime 0.2+
+        params,
+        correlation_id or _get_new_correlation_id(),
+        login_hint or "",
+        lambda result, callback_data=callback_data: callback_data.complete(result))
+    callback_data.signal.wait()
+    return _convert_result(callback_data.auth_result, client_id)
+
+
+def _acquire_token_silently(
+        authority, client_id, account_id, scopes, claims=None, correlation_id=None):
+    account = _read_account_by_id(account_id)
+    error = account.get_error()
+    if error:
+        return _convert_error(error, client_id)
+    if not account.get_account():  # It happens when the account was not created by broker
+        return
+    params = pymsalruntime.MSALRuntimeAuthParameters(client_id, authority)
+    params.set_requested_scopes(scopes)
+    if claims:
+        params.set_decoded_claims(claims)
+    callback_data = _CallbackData()
+    pymsalruntime.acquire_token_silently(
+        params,
+        correlation_id or _get_new_correlation_id(),
+        account.get_account(),
+        lambda result, callback_data=callback_data: callback_data.complete(result))
+    callback_data.signal.wait()
+    return _convert_result(callback_data.auth_result, client_id)
+
+
+def _acquire_token_interactively(
+        authority,
+        client_id,
+        account_id,
+        scopes,
+        prompt=None,  # TODO
+        claims=None,
+        extra_scopes_to_consent=None,  # TODO
+        max_age=None,  # TODO
+        correlation_id=None,
+        window=None,
+        **kwargs):
+    raise NotImplementedError("We ended up not currently using this function")
```

Comment:
>raise NotImplementedError("We ended up not currently using this function") [](http://example.com/codeflow?start=4&length=74)

May I ask why? what's the blocking issue so that we can resolve it together?

Reply:
We clarified it during our offline sync. MSAL Python's `acquire_token_silent()` api maps to MsalRuntime's `acquire_token_silently()` only. #Resolved

Reply:
Is there aplan to expose this in the future?

Reply:
> Is there a plan to expose this in the future?

For the scope and purpose of this PR, we are providing the broker behaviors inside the existing MSAL Python API/pattern, for the existing MSAL Python customers. In the future, we may evaluate when/whether to expose a different calling API/pattern.